File: mechanical/src/main/scala/resonantinduction/mechanical/gear/GearMultiBlockHandler.java
package resonantinduction.mechanical.gear;

import net.minecraft.tileentity.TileEntity;
import net.minecraftforge.common.ForgeDirection;
import resonant.lib.multiblock.MultiBlockHandler;
import universalelectricity.api.vector.Vector3;
import codechicken.multipart.TMultiPart;
import codechicken.multipart.TileMultipart;

public class GearMultiBlockHandler extends MultiBlockHandler<PartGear>
{
	public GearMultiBlockHandler(PartGear wrapper)
	{
		super(wrapper);
	}

	@Override
	public PartGear getWrapperAt(Vector3 position)
	{
		TileEntity tile = position.getTileEntity(self.getWorld());

		if (tile instanceof TileMultipart)
		{
			TMultiPart part = ((TileMultipart) tile).partMap(getPlacementSide().ordinal());

			if (part instanceof PartGear)
			{
				if (((PartGear) part).tier == self.tier)
				{
					return (PartGear) part;
				}
			}
		}

		return null;
	}

	public ForgeDirection getPlacementSide()
	{
		return self.placementSide;
	}
}

File: src/main/scala/edx/mechanical/mech/gear/GearMultiBlockHandler.scala
package edx.mechanical.mech.gear

import codechicken.multipart.TileMultipart
import net.minecraftforge.common.util.ForgeDirection
import resonant.lib.prefab.tile.multiblock.reference.MultiBlockHandler
import resonant.lib.transform.vector.Vector3

class GearMultiBlockHandler(wrapper: PartGear) extends MultiBlockHandler[PartGear](wrapper: PartGear)
{
  override def getWrapperAt(position: Vector3): PartGear =
  {
    val tile = position.getTileEntity(this.tile.getWorld)

    if (tile.isInstanceOf[TileMultipart])
    {
      val part = tile.asInstanceOf[TileMultipart].partMap(getPlacementSide.ordinal)

      if (part.isInstanceOf[PartGear])
      {
        if ((part.asInstanceOf[PartGear]).tier == this.tile.tier)
        {
          return part.asInstanceOf[PartGear]
        }
      }
    }
    return null
  }

  def getPlacementSide: ForgeDirection = tile.placementSide
}